src/structures/MessageSelectMenu.js

```diff
@@ -0,0 +1,184 @@
+
+    /**
+     * The minimum number of selections required
+     * @type {?number}
+     */
+    this.minValues = data.min_values ?? data.minValues ?? null;
+
+    /**
+     * The maximum number of selections allowed
+     * @type {?number}
+     */
+    this.maxValues = data.max_values ?? data.maxValues ?? null;
+
+    /**
+     * Options for the select menu
+     * @type {MessageSelectOption[]}
+     */
+    this.options = this.constructor.normalizeOptions(data.options ?? []);
+  }
+
+  /**
+   * Sets the custom ID of this select menu
+   * @param {string} customID A unique string to be sent in the interaction when clicked
+   * @returns {MessageSelectMenu}
+   */
+  setCustomID(customID) {
+    this.customID = Util.verifyString(customID, RangeError, 'SELECT_MENU_CUSTOM_ID');
+    return this;
+  }
+
+  /**
+   * Sets the maximum number of selections allowed for this select menu
+   * @param {number} maxValues Number of selections to be allowed
+   * @returns {MessageSelectMenu}
+   */
+  setMaxValues(maxValues) {
+    this.maxValues = maxValues;
+    return this;
+  }
+
+  /**
+   * Sets the minimum number of selections required for this select menu
+   * <info>This will default the maxValues to the number of options, unless manually set</info>
+   * @param {number} minValues Number of selections to be required
+   * @returns {MessageSelectMenu}
+   */
+  setMinValues(minValues) {
+    this.minValues = minValues;
+    return this;
+  }
+
+  /**
+   * Sets the placeholder of this select menu
+   * @param {string} placeholder Custom placeholder text to display when nothing is selected
+   * @returns {MessageSelectMenu}
+   */
+  setPlaceholder(placeholder) {
+    this.placeholder = Util.verifyString(placeholder, RangeError, 'SELECT_MENU_PLACEHOLDER');
+    return this;
+  }
+
+  /**
+   * Adds options to the select menu.
+   * @param {...(MessageSelectOption[]|MessageSelectOption[])} options The options to add
+   * @returns {MessageSelectMenu}
+   */
+  addOptions(...options) {
+    this.options.push(...this.constructor.normalizeOptions(options));
+    return this;
+  }
+
+  /**
+   * Removes, replaces, and inserts options in the select menu.
+   * @param {number} index The index to start at
+   * @param {number} deleteCount The number of options to remove
+   * @param {...MessageSelectOption|MessageSelectOption[]} [options] The replacing option objects
+   * @returns {MessageSelectMenu}
+   */
+  spliceOptions(index, deleteCount, ...options) {
+    this.options.splice(index, deleteCount, ...this.constructor.normalizeOptions(...options));
+    return this;
+  }
+
+  /**
+   * Transforms this select menu to a plain object
+   * @returns {Object} The raw data of this select menu
+   */
+  toJSON() {
+    return {
+      custom_id: this.customID,
+      placeholder: this.placeholder,
+      min_values: this.minValues,
+      max_values: this.maxValues ?? this.minValues ? this.options.length : undefined,
```

Comment:
```suggestion
      max_values: this.maxValues ?? (this.minValues ? this.options.length : undefined),
```
fixes a bug resulting in max being ignored due to operator precedence